Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM alpine:3.10
+
+RUN apk add --no-cache --update openssl bash
+
+COPY ssl-cert-check /
+COPY entrypoint.sh /
+
+# nobody:nobody
+USER 65534:65534
+
+CMD ["/entrypoint.sh"]
```

Comment:
I missed this in the initial review, but if this is going to run as a daemon, shell should not be a PID 1 - it's not designed to run as init and does not handle all the functionality that is expected from init process. A good option is to use some small init, IMHO best one is https://github.com/krallin/tini.

Reply:
Yeah, I worked with `tini` before, it is mostly useful for zombie reaping. But our daemon is single-threaded, so no need to worry about zombies, and we are now handing signals properly with entrypoint itself.

Reply:
Both `entrypoint.sh` and `ssl-cert-check` do create child processes, I recommend reading https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/.

Reply:
I am well aware of the zombie problem :) The thing is that it makes sense to worry about zombies only when process spawns A LOT of children (i.e apache with mod prefork serving hundreds requests per second), so even with a very small possibility that some amount of these children will end up zombies, they may exhaust available slots in process table. This daemon can probably process 1 request in 1-3 seconds (depending on number of certs to check), and designed to be polled once or twice daily, so IMO we should not worry about zombies here.